api/v1alpha1/openapi.yaml

```diff
@@ -2539,10 +2539,42 @@ components:
         - type: object
           properties:
             inline:
-              type: object
-              additionalProperties: true
+              type: array
+              items:
+                $ref: '#/components/schemas/FileSpec'
           required:
           - inline
+    FileSpec:
+      type: object
+      properties:
+        path:
+          type: string
+          description: The absolute path to the file on the device. Note that any existing file will be overwritten.
+        contents:
+          type: string
+          description: The URL-encoded or base64-encoded contents of the file.
+        contentsEncoding:
+          type: string
+          description: How the contents are encoded.
+          enum:
+          - base64
+          - url
+        mode:
+          type: integer
+          description: |
+            The file’s permission mode. You may specify the more familiar octal with a leading zero (e.g., 0644) or as
```

Comment:
According to the description, the decimal value should work, but it returns error 400.
POST Fleet
```
{
  "apiVersion": "v1alpha1",
  "kind": "Fleet",
  "metadata": {
    "name": "last",
    "labels": {}
  },
  "spec": {
    "selector": {
      "matchLabels": {}
    },
    "template": {
      "metadata": {
        "labels": {
          "fleet": "last"
        }
      },
      "spec": {
        "config": [
          {
            "configType": "InlineConfigProviderSpec",
            "inline": [
              {
                "path": "/etc/motd",
                "contents": "whatever",
                "contentsEncoding": "base64",
                "mode": 777,
                "user": "camadorg",
                "group": "33"
              }
            ],
            "name": "ff"
          }
        ]
      }
    }
  }
}
```

Response:
```
Error 400: Bad Request - spec.config[].inline[0].mode: Invalid value: 604: must have 3 octal digits
```

Sending the decimal value seems like the best option for us, as otherwise looks like we'd need to do some conversions for being able to send the octal value as a number.

Reply:
777 is octal, so it should be specified as 0777.

Reply:
I see, we can either convert to decimal ourselves, or send the value as `0o777` which gets converted to the decimal value. Just "0777" is not a valid value in Typescript. Either way, the API is working well, thanks for checking.

![octal](https://github.com/user-attachments/assets/a68acd08-7a5e-44bf-862b-482bbd6c04cd)